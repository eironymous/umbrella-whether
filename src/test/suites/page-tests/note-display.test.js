import React from "react";
import { mount, shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NoteElement from "../../../pages/details/detail-card-elements/notes/note-element";
import NoteInput from "../../../components/note-input";

configure({ adapter: new Adapter() });

const note = {
	body: "This is a test note. It has some things in it.",
	timeStamp: "2020/09/23 04:14:00"
};

test('NoteElement loads', () => {
	let wrapper = mount(
		<NoteElement note={note} />
	);

	expect(wrapper).toMatchSnapshot();
});

test('NoteElement returns null if note is undefined', () => {
	let wrapper = shallow(
		<NoteElement note={undefined} />
	);

	expect(wrapper.get(0)).toBeFalsy();
});

test('NoteElement populates body correctly', () => {
	let wrapper = mount(
		<NoteElement note={note} />
	);

	expect(wrapper.find(".note-body-text").first().text()).toContain(note.body);
});

test('NoteElement populates timestamp correctly', () => {
	let wrapper = mount(
		<NoteElement note={note} />
	);

	expect(wrapper.find(".note-timestamp-text").first().text()).toContain(note.timeStamp);
});

test('NoteInput populates text correctly', () => {
	let wrapper = mount(
		<NoteInput text={note.body} />
	);

	expect(wrapper.find(".note-input").first().text()).toContain(note.body);
});