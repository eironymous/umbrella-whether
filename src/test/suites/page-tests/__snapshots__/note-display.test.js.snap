// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteElement loads 1`] = `ReactWrapper {}`;
